metricbeat/helper/http.go

```diff
@@ -74,8 +78,8 @@ func NewHTTP(base mb.BaseMetricSet) (*HTTP, error) {
 
 	var dialer, tlsDialer transport.Dialer
 
-	dialer = transport.NetDialer(config.Timeout)
-	tlsDialer, err = transport.TLSDialer(dialer, tlsConfig, config.Timeout)
+	dialer = transport.NetDialer(config.ConnectTimeout)
+	tlsDialer, err = transport.TLSDialer(dialer, tlsConfig, config.ConnectTimeout)
```

Comment:
Do we need to improve the error message to make clear what timeout we hit?

Reply:
Error messages for these timeouts are generated by the networking libraries, it wouldn't be so straight-forward to interpret them.

They can be differentiated though, request timeout says something like `net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

And connection timeout is an `i/o timeout`.

If you think it can be confusing to have two different timeouts or low level errors, I am fine with leaving just the old request `Timeout`.

Reply:
I like the separation. 👍